src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -436,6 +487,9 @@ public void writeTo(StreamOutput out) throws IOException {
         if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(out.getVersion(), IndexUtil.MODEL_METHOD_COMPONENT_CONTEXT_KEY)) {
             getMethodComponentContext().writeTo(out);
         }
+        if (IndexUtil.isVersionOnOrAfterMinRequiredVersion(out.getVersion(), KNNConstants.MODEL_VECTOR_DATA_TYPE_KEY)) {
+            out.writeOptionalString(vectorDataType.getValue());
```

Comment:
I see vectorDataType is non null because we assign Float as default. Then, this line doesn't have to be optionalString but simply writeString();
```suggestion
            out.writeString(vectorDataType.getValue());
```

Reply:
ack